src/tribler/core/components/libtorrent/tests/test_download_manager.py

```diff
@@ -455,3 +492,71 @@ async def test_check_for_dht_ready(fake_dlmgr):
     fake_dlmgr.get_session().status().dht_nodes = 1000
     # If the session has enough peers, it should finish instantly
     await fake_dlmgr._check_dht_ready()
+
+
+def test_update_trackers(fake_dlmgr: DownloadManager) -> None:
+    fake_download, dl_state = create_fake_download_and_state()
+    fake_dlmgr.downloads[fake_download.infohash] = fake_download
+    fake_metainfo = {b'info': {b'name': b"test_download"}}
+    fake_download.get_def().metainfo = fake_metainfo
+
+    fake_tracker1 = "127.0.0.1/test-announce1"
+
+    fake_dlmgr.update_trackers(fake_download.infohash, [fake_tracker1])
+
+    assert fake_metainfo["announce"] == fake_tracker1
+    assert "announce-list" not in fake_metainfo
+
+
+def test_update_trackers_list(fake_dlmgr: DownloadManager) -> None:
+    fake_download, dl_state = create_fake_download_and_state()
+    fake_dlmgr.downloads[fake_download.infohash] = fake_download
+    fake_metainfo = {b'info': {b'name': b"test_download"}}
+    fake_download.get_def().metainfo = fake_metainfo
+
+    fake_tracker1 = "127.0.0.1/test-announce1"
+    fake_tracker2 = "127.0.0.1/test-announce2"
+
+    fake_dlmgr.update_trackers(fake_download.infohash, [fake_tracker1, fake_tracker2])
+
+    assert "announce" not in fake_metainfo
+    assert fake_metainfo["announce-list"] == [[fake_tracker1, fake_tracker2]]
```

Comment:
Apologies, my bad: on different operating systems (win + mac apparently) the set unwrapping is different.

The output can be either `[[fake_tracker1, fake_tracker2]]` **or** `[[fake_tracker2, fake_tracker1]]`. The same is true for the next test.

Reply:
Is the following code acceptable?
```py
assert (
    fake_metainfo["announce-list"] == [[fake_tracker1, fake_tracker2]] or
    fake_metainfo["announce-list"] == [[fake_tracker2, fake_tracker1]]
)
```

Reply:
Looks good to me 👍

Reply:
Which option? I added one in an edit to my message above.

Reply:
Ah, I see. 👀 I was referring to the original code you posted with the `or`: the first option. I'd rather avoid checking the Operating System (unless absolutely no other options exist).

Reply:
Will do. Thanks 😃